google-cloud-storage/src/main/java/com/google/cloud/storage/BlobAppendableUpload.java

```diff
@@ -90,14 +91,38 @@ public interface BlobAppendableUpload extends BlobWriteSession {
    * <p>This interface allows writing bytes to an Appendable Upload, and provides methods to close
    * this channel -- optionally finalizing the upload.
    *
+   * <p>The {@link #write(ByteBuffer)} method of this channel is non-blocking.
+   *
    * @since 2.51.0 This new api is in preview and is subject to breaking changes.
    */
   @BetaApi
   @InternalExtensionOnly
   interface AppendableUploadWriteableByteChannel extends WritableByteChannel {
 
     /**
-     * Finalize the upload and close this instance to further {@link #write(ByteBuffer)}ing. This
+     * <b>This method is non-blocking</b>
+     *
+     * <p>Consume as many bytes as can fit in the underlying outbound queue. The size of the
+     * outbound queue is determined from {@link BlobAppendableUploadConfig#getFlushPolicy()}{@code
+     * .}{@link FlushPolicy#getMaxPendingBytes() getMaxPendingBytes()}. If the outbound queue is
+     * full, and can not fit more bytes, this method will return 0.
+     *
+     * <p>This method may be invoked at any time. If another thread has already initiated a write
+     * operation upon this channel, however, then an invocation of this method will block until the
+     * first operation is complete.
```

Comment:
Seems contradictory with the first claim? Unless we don't consider that blocking because the first call is simply doing a memory copy?

Reply:
I'll remove that. We can make the lock acquisition conditional too and return 0 if we can't acquire the lock.